### ISSUE TITLE
encoding pdf417_compact has extra white space

### ISSUE THREAD

**Issue by agkyunromb** [rxing-core/rxing#31]
### my code
~~~rust
use rxing::helpers::save_file;
use rxing::common::BitMatrix;
use rxing::pdf417::encoder::Dimensions;
use rxing::pdf417::PDF417Writer;
use rxing::BarcodeFormat::PDF_417;
use rxing::EncodeHintType::{ERROR_CORRECTION, MARGIN, PDF417_COMPACT, PDF417_DIMENSIONS};
use rxing::EncodeHintValue::{ErrorCorrection, Margin, Pdf417Compact, Pdf417Dimensions};
use rxing::EncodingHintDictionary as Hint;
use rxing::Writer;

fn main(){
    let ring = ring_code("test", 103, 270);
    save_file("test.jpg", &ring).unwrap();
}


fn ring_code(contents: &str, width: i32, height: i32) -> BitMatrix {
    let writer = PDF417Writer::default();
    let dimensions = Dimensions::new(2, 2, 70, 70);
    let mut hints = Hint::new();
    hints.insert(ERROR_CORRECTION, ErrorCorrection("6".to_string()));
    hints.insert(PDF417_DIMENSIONS, Pdf417Dimensions(dimensions));
    hints.insert(PDF417_COMPACT, Pdf417Compact("true".to_string()));
    hints.insert(MARGIN, Margin(("5").to_string()));

    writer
        .encode_with_hints(contents, &PDF_417, width, height, &hints)
        .unwrap()
}
~~~

### result
![test](https://github.com/rxing-core/rxing/assets/26702058/7c6365b1-9f86-4fff-899b-859cd4c075b2)
as you can see it have extra white space
### it should be
![test](https://github.com/rxing-core/rxing/assets/26702058/dd64674f-6b53-453f-8022-5a0ca4a1f368)

#34

**Comment by hschimke** [comment]
Fixed by #34 fixed this issue

**Comment by hschimke** [comment]
This fix appears in v0.4.6, currently uploading to cargo
